File: 26.remove-duplicates-from-sorted-array/solution.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 0;
        for(int i=0; i<nums.size(); i++) {
            count++;
            while(nums[i+1]==nums[i]) {
                i++;
            }
            nums[count] = nums[i+1];
        }
        return count;
    }
};

File: 26.remove-duplicates-from-sorted-array/solution.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut i = 0;
        let mut j = 0;
        while j < nums.len() {
            if nums[i] == nums[j] {
                j+= 1;
            } else {
                i+= 1;
                nums[i] = nums[j];
            }
        }
        (i+1) as i32
    }
}
